back/app/controllers/omniauth_callback_controller.rb

```diff
@@ -134,12 +134,23 @@ def set_auth_cookie(provider: nil)
     }
   end
 
+  # Updates the user with attributes from the auth response if `updateable_user_attrs` is set
+  # Overwrites current attributes by default unless `overwrite_attrs?` is set to false on the authver method
   def update_user!(auth, user, authver_method)
     return unless authver_method.respond_to? :updateable_user_attrs
 
     attrs = authver_method.updateable_user_attrs
+    overwrite_attrs = authver_method.respond_to?(:overwrite_user_attrs?) && authver_method.overwrite_user_attrs?
```

Comment:
We add more and more `authver_method.respond_to?` in our code (I'll need to add one more in my FranceConnect PR). In the end, it will be really hard to see what methods can be / are implemented on each `authver_method`. There's no schema/structure for them.

Do you think it would be useful to implement e.g. `BaseAuthverMethod` with some default implementations of all these methods? E.g., 

```ruby
def updateable_user_attrs
  []
end
```

If yes, I could do it in my PR.

Reply:
And also
```ruby
    def overwrite_user_attrs?
      true
    end
```

Reply:
I thought about the very same thing actually. Maybe lets' have a base class to inherit from and make the interface explicit that we are expecting here. I can implement that tomorrow on my branch. Great to see that you are having similar feelings about the code. ^.^